From d7113b9e9bb08f4055af2a4bdf46cedd515265f7 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Mon, 15 Jun 2015 18:26:37 +0800
Subject: [PATCH 0761/5242] MLK-11115 arm: imx: double check m4 is enabled

commit  aa599269ba1a2cb3acd219d1951405c27b87e7b9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Make a double check that m4 is enabled and run on imx7d

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
(cherry picked from commit 276fe0947a13c74dad003b0fc3735d389928c1e3)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/src.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/src.c b/arch/arm/mach-imx/src.c
index 5f9651f..e410998 100644
--- a/arch/arm/mach-imx/src.c
+++ b/arch/arm/mach-imx/src.c
@@ -170,7 +170,7 @@ void __init imx_src_init(void)
 
 	if (cpu_is_imx7d()) {
 		val = readl_relaxed(src_base + SRC_M4RCR);
-		if ((val & BIT(3)) == BIT(3))
+		if (((val & BIT(3)) == BIT(3)) && !(val & BIT(0)))
 			m4_is_enabled = true;
 		else
 			m4_is_enabled = false;
-- 
1.7.9.5

